text/0012-random.md

```diff
@@ -0,0 +1,128 @@
+- Feature Name: Source of Entropy
+- Start Date: 2024-07-15
+- RFC PR: [twizzler-rfcs/rfcs#0012](https://github.com/twizzler-operating-system/rfcs/pull/0009)
+- Twizzler Issue: [twizzler-operating-system/twizzler#0011](https://github.com/twizzler-operating-system/twizzler/issues/0009)
+
+# Summary
+
+[summary]: #summary
+
+Adds a secure source of entropy to the kernel, similar to /dev/random on Twizzler. It should seed itself at startup with a true random number generator (TRNG) and then use a Cryptographically secure Pseudo Random Number Generator (CSPRNG) to generate following requests for randomness.
+
+# Motivation
+
+[motivation]: #motivation
+
+This RFC is useful because randomness is used for cryptographic operations and the kernel certainly requires cryptographic operations such as private/public key generation and UUID generation for the twizzler security model. It is also very likely userspace programs will also require entropy for userspace cryptographic operations (or games or other applications which require randomness).
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+The module should expose a `getrandom(dest: &mut [u8])` function (and syscall) which will entirely fill the `dest` buffer with bytes. It might block while doing so during the initial seeding of the CSPRNG. I also propose providing a `getrandom_nonblocking(dest: &mut [u8]) -> usize` which will do the same but without blocking, returning the number of bytes overwritten into `dest` (which can be 0).
```

Comment:
I'd suggest we do getrandom(dest: &mut [u8], flags: GetRandomFlags), where flags would be defined using the bitflags crate. For now, we'd only have one, "NONBLOCKING". Allows future additions if we reserve all other flags for future expansion.